lib/puppet/provider/snmp_notification/cisco_ios.rb

```diff
@@ -44,10 +44,31 @@ def self.commands_from_instance(property_hash)
       commands_array
     end
 
-    def get(context, _names = nil)
+    def get(context, names = [])
       output = context.transport.run_command_enable_mode(PuppetX::CiscoIOS::Utility.get_values(commands_hash))
       return [] if output.nil?
       return_value = Puppet::Provider::SnmpNotification::CiscoIos.instances_from_cli(output)
+      unless names.empty?
+        # as the `enable` property is like an
+        # ensure except it does not get returned
+        # if the trap is not enabled
+        # we create our own return value
+        # when the return values from system
+        # does not contain the name requested, i.e.
+        # in an apply - this will allow the provider to be
+        # idempotent on both a `true` and `false` snmp_notification
+        # resource
```

Comment:
This took me way to long to figure out what that comment actually means.

Here's what I'm thinking you've been trying to say:

```
# When trying to disable snmp_notifications that are already disabled, puppet does not create a proper
# `ensure => absent` resource, because netdev_stdlib types are not ensurable.
# This leads to annoying/buggy "Snmp_notification[X]: changed 'enable' property from  to 'false'"
# messages and useless calls to the provider.
# If we get passed a `name` we can work around this by synthesising a disabled resource.
```

* Describe the situation when a problem is happening
* Describe the symptoms (log messages etc) and consequences (provider calls)
* A short summary of the solution

The code will have to change to deal with `names == nil` (return all existing notifications).